### PR TITLE
feat(TDP-1712): updating tests with new configurator2

### PR DESCRIPTION
**Pull request by DMackintosh-TNL**
Welcome to the jungle, we got fun and games...

Recreated my branch as I have inadvertently merged with with the old base, and broken it...

### CODE REVIEW

File: packages/slice-layout/__tests__/jest.config.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
-const jestConfigurator = require("@times-components/jest-configurator").default;
+const jestConfigurator = require("@times-components/jest-configurator-web")
+  .default;
 
-module.exports = jestConfigurator("web", __dirname, {
+module.exports = jestConfigurator(__dirname, {
   coverageIgnoreGlobs: ["index.js", "slice-layout.js", "styles.js"]
```
Comment (bibicollins1) [inline packages/slice-layout/__tests__/jest.config.js:5]: Wondering whether we can just remove this bit? Not really doing coverage atm...

File: packages/star-button/package.json [hunk 2]
```diff
@@ -35,8 +32,10 @@
...
   "devDependencies": {
     "@babel/core": "7.4.4",
+    "@babel/preset-env": "7.4.4",
+    "@babel/preset-react": "7.9.1",
+    "@times-components/jest-configurator-web": "0.7.0",
     "@times-components/eslint-config-thetimes": "0.8.18",
-    "@times-components/jest-configurator": "2.8.0",
     "@times-components/jest-serializer": "3.3.0",
     "@times-components/storybook": "4.4.0",
     "@times-components/test-utils": "2.3.10",
```
Comment (bibicollins1) [inline packages/star-button/package.json:41]: React-native dependencies still in package.json

File: packages/tealium/package.json [hunk 3]
```diff
@@ -27,8 +27,10 @@
...
   "devDependencies": {
     "@babel/core": "7.4.4",
+    "@babel/preset-env": "7.4.4",
+    "@babel/preset-react": "7.9.1",
+    "@times-components/jest-configurator-web": "0.7.0",
     "@times-components/eslint-config-thetimes": "0.8.18",
-    "@times-components/jest-configurator": "2.8.0",
     "@times-components/test-utils": "2.3.10",
     "@times-components/webpack-configurator": "2.0.29",
     "babel-jest": "24.8.0",
```
Comment (bibicollins1) [inline packages/tealium/package.json:36]: Still have react-native dependencies in package.json

File: packages/test-utils/package.json [hunk 4]
```diff
@@ -42,6 +42,11 @@
     "opentype.js": "1.1.0"
   },
   "devDependencies": {
+    "@babel/cli": "7.4.4",
```
Comment (bibicollins1) [inline packages/test-utils/package.json:45]: Not sure this is used?

File: packages/tracking/__tests__/jest.config.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
-const jestConfigurator = require("@times-components/jest-configurator").default;
+const jestConfigurator = require("@times-components/jest-configurator-web")
+  .default;
 const path = require("path");
 
-module.exports = jestConfigurator("web", __dirname, {
+module.exports = jestConfigurator(__dirname, {
   coverageIgnoreGlobs: ["tracking-context-types.js", "tracking.js"],
```
Comment (bibicollins1) [inline packages/tracking/__tests__/jest.config.js:6]: Not sure we need these, as should be covered in new config.

File: packages/tracking/__tests__/jest.config.js [hunk 6]
```diff
@@ -1,7 +1,8 @@
-const jestConfigurator = require("@times-components/jest-configurator").default;
+const jestConfigurator = require("@times-components/jest-configurator-web")
+  .default;
 const path = require("path");
```
Comment (bibicollins1) [inline packages/tracking/__tests__/jest.config.js:4]: Can't comment on the file, but look into why rename of resolve-attrs.test.js to resolve-attrs-test.js

File: packages/tracking/__tests__/tracking.test.js [hunk 7]
```diff
@@ -1,474 +0,0 @@
-import React, { Component } from "react";
```
Comment (bibicollins1) [inline packages/tracking/__tests__/tracking.test.js:1]: Unsure where this has been moved to?

File: packages/slice-layout/__tests__/jest.config.js (hunk 1 above)
Comment (adamosborne-tnl) [reply packages/slice-layout/__tests__/jest.config.js:5]: yeah, all coverageIgnoreGlobs can go as wasn't doing anything

File: packages/user-state/package.json [hunk 8]
```diff
@@ -41,8 +38,10 @@
...
   "devDependencies": {
     "@babel/core": "7.4.4",
+    "@babel/preset-env": "7.4.4",
+    "@babel/preset-react": "7.9.1",
+    "@times-components/jest-configurator-web": "0.7.0",
     "@times-components/eslint-config-thetimes": "0.8.18",
-    "@times-components/jest-configurator": "2.8.0",
     "@times-components/jest-serializer": "3.3.0",
     "@times-components/webpack-configurator": "2.0.29",
     "babel-jest": "24.8.0",
```
Comment (bibicollins1) [inline packages/user-state/package.json:47]: Still have react-native dependencies

File: packages/video/package.json [hunk 9]
```diff
@@ -8,7 +8,7 @@
     "prettier:diff": "prettier --list-different '**/*.*'",
     "depcheck": "depcheck --ignores='@babel/*,babel-*,depcheck,eslint,jest,prettier,react-art,webpack*' --ignore-bin-package=false --skip-missing",
     "lint": "eslint . && yarn prettier:diff && yarn depcheck",
-    "test:web": "jest --config='./__tests__/web/jest.config.js'",
+    "test:web": "jest --config='./__tests__/jest.config.js'",
     "prepublishOnly": "yarn transpile && yarn bundle",
     "cleanup-dist": "rm -rf dist",
     "transpile": "yarn cleanup-dist && babel src -d dist",
```
Comment (bibicollins1) [inline packages/video/package.json:14]: Can remove local jest configuration

File: packages/star-button/package.json (hunk 2 above)
Comment (adamosborne-tnl) [reply packages/star-button/package.json:41]: ok, removing

Comment (bibicollins1) [reply packages/star-button/package.json:41]: @adamosborne-tnl I'm happy to jump on some of these also sorry was more notes for myself to look into :)

File: packages/tealium/package.json (hunk 3 above)
Comment (adamosborne-tnl) [reply packages/tealium/package.json:36]: ok. removing

File: packages/test-utils/package.json (hunk 4 above)
Comment (adamosborne-tnl) [reply packages/test-utils/package.json]: it's not, removing

File: packages/tracking/__tests__/jest.config.js (hunk 5 above)
Comment (adamosborne-tnl) [reply packages/tracking/__tests__/jest.config.js]: nope, removing

File: packages/tracking/__tests__/jest.config.js (hunk 6 above)
Comment (adamosborne-tnl) [reply packages/tracking/__tests__/jest.config.js:4]: resolve-attrs is included in tracking.web.test and would run twice without rename, so leaving

File: packages/user-state/package.json (hunk 8 above)
Comment (adamosborne-tnl) [reply packages/user-state/package.json:47]: ok, removing

File: packages/video/package.json (hunk 9 above)
Comment (adamosborne-tnl) [reply packages/video/package.json:14]: ok. removed

File: packages/tracking/__tests__/tracking.test.js (hunk 7 above)
Comment (adamosborne-tnl) [reply packages/tracking/__tests__/tracking.test.js:1]: has been changed to tracking-test.js

Comment (adamosborne-tnl) [reply packages/tracking/__tests__/tracking.test.js:1]: is still imported as part of other tests. 👍

File: packages/test-utils/package.json [hunk 10]
```diff
@@ -42,6 +42,10 @@
     "opentype.js": "1.1.0"
   },
   "devDependencies": {
+    "@babel/core": "7.4.4",
+    "@babel/preset-env": "7.4.4",
+    "@babel/preset-react": "7.9.1",
+    "@babel/plugin-transform-runtime": "7.4.4",
```
Comment (bibicollins1) [inline packages/test-utils/package.json:48]: Do we use this here?

File: packages/tracking/__tests__/jest.config.js (hunk 6 above)
Comment (bibicollins1) [reply packages/tracking/__tests__/jest.config.js:4]: Ah okay nice 👍 Thanks!

File: packages/tracking/tracking.showcase.js [hunk 11]
```diff
@@ -10,19 +10,30 @@ import {
...
-  <View style={[boxStyles.box, { backgroundColor: color }]}>
-    <Button onPress={() => onPress("button 1")} title="Press me" />
-    <Button
-      color="limegreen"
-      onPress={() => onPress("button 2")}
-      title="Or me"
-    />
-  </View>
+const BoxWithButtons = ({ color, onClick }) => (
+  <TcView>
```
Comment (bibicollins1) [inline packages/tracking/tracking.showcase.js:14]: wondering why two wrapping <TcViews>?

File: packages/test-utils/package.json (hunk 10 above)
Comment (adamosborne-tnl) [reply packages/test-utils/package.json:48]: i think so, the babel.config has been updated to remove the metro-react

File: packages/tracking/tracking.showcase.js (hunk 11 above)
Comment (adamosborne-tnl) [reply packages/tracking/tracking.showcase.js:14]: i don't see the need. will remove extra one

**Review by bibicollins1 — APPROVED** [review]

**Pull request merged**